### DISCUSSION TITLE
Difficulty installing Firedrake in Kubuntu

### DISCUSSION THREAD

**Discussion by AchillesMavr** [firedrakeproject/firedrake#3384] [Firedrake support] [answered]
I had downloaded firedrake in the same system about 2 years ago and everything was fine. Few days back I decided to update it and I got a bunch errors. Unable to resolve all the errors, I uninstalled  and tried to install it again.

There I again got an error message at end regarding the libsupermesh package, as of now it seems I need Cmake 3.18 and higher, and I currently run Kubuntu 20.04 which is based on Cmake 3.16.3.

My question is since I do not want to mess with Cmake, is there any chance  that there is an previous firedrake-install script needing 3.16 Cmake?

**Comment by dham** [comment, marked as answer]
It's not the install script which uses CMake, it's various compiled dependencies that the install script installs. Using an old install script isn't going to wind back those dependencies. Simply installing a more recent CMake is the easiest way out of your current issue.

**Reply by AchillesMavr** [reply]
Thank you for your reply. I see, it makes more sense. I will try updating CMake

**Reply by francesco-ballarin** [reply]
You may want to keep in mind that kitware offers an apt repository at https://apt.kitware.com/, I am quite sure that with that you would easily install a recent `cmake` version.

**Reply by AchillesMavr** [reply]
Thank you so much, that's quite helpful. Do I  still need to uninstall the version installed with Kubuntu?

**Reply by francesco-ballarin** [reply]
No. If everything works correctly, you should be asked to upgrade your existing version 3.16.3 to the one currently available in the kitware repo.

**Reply by AchillesMavr** [reply]
Thank you again. I will try this.
